### PR TITLE
Add planout to vue-phoenix

### PR DESCRIPTION
**Pull request by rammie**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,43 +11,45 @@
     "axios": ">= 0.18.0",
     "jquery": "^3.3.1",
     "lodash": "^4.17.11",
+    "planout": "git://github.com/rawls238/Planout.js.git#97807c149d01031b972d9c3df7fb93558a166120",
```
Comment (mohanyin) [inline package.json:14]: Any chance we can move onto a normal version of Planout now that you added type definitions?

File: package.json [hunk 2]
```diff
@@ -11,43 +11,45 @@
...
-    "eslint-plugin-no-only-tests": "^2.4.0",
-    "eslint-plugin-node": "^11.0.0",
-    "eslint-plugin-promise": "^4.2.1",
-    "eslint-plugin-standard": "^4.0.1",
-    "eslint-plugin-vue": "^6.2.2",
+    "@types/lodash": "4.14.121",
+    "@typescript-eslint/eslint-plugin": "2.25.0",
+    "@typescript-eslint/parser": "2.25.0",
+    "@typescript-eslint/typescript-estree": "2.25.0",
+    "@vue/cli-plugin-eslint": "4.2.3",
```
Comment (mohanyin) [inline package.json:26]: Why are we downgrading the vue CLI + plugins?

File: src/plugins/planout.ts [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+export const planout = new PlanoutPlugin();
+
+export default (Vue: typeof _Vue, options: PlanoutPluginOptions): void => {
+  planout.setOptions(options);
+
+  Object.defineProperties(Vue.prototype, {
+    $planout: {
+      get() {
+        planout.fetchConfig();
```
Comment (mohanyin) [inline src/plugins/planout.ts:359]: I'm a bit confused by this line. Does this say that every time we do `this.$planout.<whatever>`, it'll fetch the planout config?

File: src/plugins/planout.ts [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+  }
+
+  async fetchConfig() {
+    try {
+      const response = await this.axios.get<PlanoutConfig>(`/cms/planout/config/${options.appName}`);
+      const config = await options.getConfig();
+      if (!isEqual(response.data, config)) {
+        console.info('Planout: Experiment config updated. Re-initializing.');
+        options.setConfig(response.data);
+        this.useConfig(response.data);
```
Comment (mohanyin) [inline src/plugins/planout.ts:341]: A bit confused about this line as well. Wouldn't we always want to `useConfig` when initializing the plugin?

File: tests/unit/planout.spec.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    const counts: { [key: string]: number } = {};
+    for (let i = 0; i < 100; i++) {
+      planout.setInput('uid', i.toString());
+      const value = planout.get('ns1', 'variable1', 'gohan');
+      if (!counts[value as string]) {
+        counts[value as string] = 0;
+      }
+      counts[value as string] += 1;
+    }
+    expect(counts.goku).toBeGreaterThan(45);
```
Comment (mohanyin) [inline tests/unit/planout.spec.ts:122]: Is this process deterministic or are we relying on probability here?

File: package.json (hunk 2 above)
Comment (rammie) [reply package.json:26]: This combination removed all warnings.

File: package.json (hunk 1 above)
Comment (rammie) [reply package.json:14]: This is the current tip of the normal version, the maintainer has not yet uploaded the new package to npm after merging the PR.
See:
https://github.com/rawls238/PlanOut.js/commits/master 
https://www.npmjs.com/package/planout

File: tests/unit/planout.spec.ts (hunk 5 above)
Comment (rammie) [reply tests/unit/planout.spec.ts:122]: It's deterministic

File: package.json (hunk 1 above)
Comment (mohanyin) [reply package.json:14]: oh whoops misread that... thought we were using a fork

File: tests/unit/planout.spec.ts (hunk 5 above)
Comment (mohanyin) [reply tests/unit/planout.spec.ts:122]: ah got it. i think it's worth a comment or a more specific assertion

File: src/plugins/planout.ts (hunk 4 above)
Comment (mohanyin) [reply src/plugins/planout.ts:341]: Still confused... i can't find where this is called now

Comment (rammie) [reply src/plugins/planout.ts]: Refactored (well you did)

**Pull request merged**